app/routes.py

```diff
@@ -1,2 +1,289 @@
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_task():
+
+    # tasks hold all the information from the database. Becasue Task is referring to a table. So we are capturing all the records(rows) in the table and holding it in tasks.
+    # queries are the records (i.e. rows)
+    # creating task_response variable
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    task_responses = []
+    for task in tasks:
+        task_response = {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+        if task.goal_id:
+            task_response["goal_id"] = task.goal_id
+        task_responses.append(task_response)
+    return jsonify(task_responses), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    task_response = {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
+    if task.goal_id:
+        task_response["goal_id"] = task.goal_id
+
+    response_body = {"task": task_response}
+
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify({
+        "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+    }), 200
+
+
+def slack_bot(title):
+    query_path = {'channel': 'slack_api_test_channel', 'text': title}
+    header = {'Authorization': os.environ.get('BOT')}
```

Comment:
From the way you set up the Authorization header, it looks like you might have included the `Bearer` string in your `.env` token definition? While this makes the setting easier here, I would tend to have the `Bearer` portion set here, and it's not really a part of the token itself. It says how to use the token. It's a minor distinction, but one worth keeping in mind. Another way that the token can be supplied is as part of the post body, in which case it _doesn't_ need to have `Bearer` included in the data. Consider that in a larger application, we might be using a token at several places in a system, and each might be passing it a different way, so having the "one true" token in the config allows us to use it however we need to elsewhere. Also, simply from a configuration standpoint, it's easy to explain to someone that to get the slack feature to work, they need to paste their token from Slack into their config file or environment variable rather than explaining that they need to prefix the token with `Bearer`.